ckan/lib/helpers.py

```diff
@@ -868,7 +868,11 @@ def dict_list_reduce(list_, key, unique=True):
     values for the key with unique values if requested. '''
     new_list = []
     for item in list_:
-        value = item.get(key)
+        try:
+            value = item.get(key)
+        except AttributeError:
+            # item might not be a dict.
+            continue
```

Comment:
This is weird, perhaps @amercader might know if this is a dcat extension specific weirdness.

As a temporary workaround, you can probably get away with (obviously untested)..

```
h.dict_list_reduce(package.resources.values(), 'format')
```

but it should only be a temporary workaround.